enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,306 @@
+undesirable for this feature.  
+- The MCO API has not been intuitive for the type of users to date who are interested in introducing and then consuming
+small collections of files on a host.
+
+This proposal will describe a [CSI driver](https://github.com/container-storage-interface/spec/blob/master/spec.md) based
+means of OpenShift cluster administrators the means of injecting K8S Secrets onto each host/node in an OpenShift cluster, 
+and then allow owners/authors/users of Pods to opt into the consumption of the content of those Secrets during their 
+Pod's execution.  The means of opting in will entail setting an annotation or label on an object vs. explicitly
+defining the volume related API objects or volume plug points into the Pod.
+
+This proposal was broken off from [a proposal on injecting RHEL entitlements/subscriptions](https://github.com/openshift/enhancements/pull/214)
+**EDITORIAL NOTE:** when PR 214 merges the reference link will be adjusted to point to the merged document hosted on
+https://github.com/openshift/enhancements.
+
+
+## Motivation
+
+Simplify overall usability for injecting and consuming Secret content onto the OpenShift hosts/nodes for use in user
+Pods.
+
+### Goals
+
+Provide alternatives to users specifying privileged Pods with HostPath volume.
+
+Provide alternatives to cluster administrators for using the MCO for injecting content onto nodes.
+
+### Non-Goals
+
+1. This proposal assumes the sensitive content encapsulated in the K8S Secret is valid and correct for the user's purposes.
+No validation of the content for specific usage within a Pod will be done.  That said, this proposal does provide 
+for watching for updates to the K8S Secrets being stored on the host/node by the associated CSI Driver of this proposal.
+So if the cluster administrator realized on their own they have provided bad data, and then corrects it, this proposal
+will watch for, discover, and propagate the update to the host/node filesystem.
+
+2. Protecting this content from being viewed/copied by users. If it's available for the Pod to use, the user who can 
+create a Pod can also read / view / exfiltrate the Secret contents. It is technically impossible to avoid that.
+
+3. 4.x customer feedback has also revealed a desire for namespace level granularity when injecting/consuming
+K8S Secret information available on the underlying host/node.  While this proposal will articulate some thought on 
+how to do this, ultimately a solution based on a given Secret utilized across multiple namespaces is deemed "meets
+minimum" for this proposal, and per namespace secret injection is out of scope.   
+
+## Proposal
+
+### User Stories [optional]
+
+1. As a cluster admin with users who needs to have their Pod access a finite set of content on a host/node in read only 
+fashion without marking their Pod privileged, I want to provide my users a means of doing that.
```

Comment:
I agree - these user stories are a bit abstract. IMO concrete use cases can help clarify this proposal, such as:

As a cluster admin with an OpenShift subscription
I want to share my entitlement keys and configuration with my workload pods
So that containers in my pods can install RHEL subscription content

Reply:
So my perspective on preferences around which details to put in which proposals, similar to what I noted in https://github.com/openshift/enhancements/pull/250#discussion_r397358585 , surfaces here as well again from my perspective.

And granted, some nuances of reviewer's suggestions may have become lost on me with the high, iterative churn.

So to be clear, and I think I agree if this is where you are coming from, for `User Stories` @adambkaplan @bparees :
- we are comfortable with specifics like the entitlement use case, even though it will be addressed in a broader fashion elsewhere?
- is this proposal allowed to only list a subset of those use cases, or does it need to be kept up to date as things evolve
- Or do we just want to put a ref to other proposal(s) where the scenarios are articluated ?

thanks

Reply:
> we are comfortable with specifics like the entitlement use case, even though it will be addressed in a broader fashion elsewhere?

i'm comfortable with entitlements as a specific example of a use case.  The details of the proposal itself should not be tied to entitlements, they should just be sufficient to see that the proposal, when implemented, will address that use case (among others)

Another specific example of a use case worth including is distribution of CAs provided by the admin, to workloads.

> is this proposal allowed to only list a subset of those use cases, or does it need to be kept up to date as things evolve

you don't have to list all the ways people might use it, just enough to motivate why we need to do this.